tools/serverpod_cli/test/generator/open_api/excepted_file_created_test.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:serverpod_cli/analyzer.dart';
+import 'package:serverpod_cli/src/analyzer/dart/definitions.dart';
+import 'package:serverpod_cli/src/generator/open_api/open_api_generator.dart';
+import 'package:serverpod_cli/src/test_util/builders/generator_config_builder.dart';
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+
+const projectName = 'example_project';
+var config = GeneratorConfigBuilder().withName(projectName).build();
+const generator = OpenApiGenerator();
+void main() {
+  group('Given a single Endpoint and Entity when generating openapi schema',
```

Comment:
And just to give some extra context to what I mean a test could look like this:

```dart
YamlMap extractYamlContent(String fileContent) {
  YamlDocument document = loadYamlDocument(fileContent);
  return document.contents as YamlMap;
}
```

```dart
void main() {
  var expectedFilePath =
      path.join('generated', 'openapi', 'openapi.json');

  test(
      'Given an endpoint with a documentation comment without whitespace when generating the openAPI spec then the /// is removed.',
      () {
    var endpoints = [
      EndpointDefinitionBuilder()
          .withDocumentationComment('///A documentation comment')
          ...
          .build(),
    ];

    var entities = [
      ClassDefinitionBuilder()
          ...
          .build(),
    ];

    ProtocolDefinition protocolDefinition = ProtocolDefinition(
      endpoints: endpoints,
      entities: entities,
    );

    var codeMap = generator.generateOpenAPISchema(
        protocolDefinition: protocolDefinition, config: config);
      
    var openApiSchema = extractYamlContent(codeMap[expectedFilePath]!);

    // An example of how to access the key, replace this with the real path to test.
    var description = (openApiSchema.nodes['paths'] as YamlMap).nodes['description'];

    expect(description, contains('A documentation comment'));
  });
}
```

Or you could wrap this in a `group` and reuse the setup where it would make sense. Point here is that we are testing a very limited thing within the document based on data that looks and behaves as what we will have in the real world.

We get a few benefits by testing in this way.

1. The output is always verified to be valid yaml syntax.
2. We test that our input generates the correct thing.
3. We test on a higher abstraction level which means we are free to modify the production however we want without having to modify this test.

Reply:
If there is something that is unclear here please just ask! :)

Reply:
I've made updates to all the tests. Can you please review.